metatron-app/metatron-discovery
# fix Quartz initial script

### Description
An error occurs when applying ChangeSet about Quartz based on MySQL to H2.
Quartz schema creation scripts vary by database.
Modified to execute different sql by database(H2, MySQL, Postgesql, Oracle, MSSQL).

**Related Issue** : 

### How Has This Been Tested?
1. Delete existing H2 database if it exists.
2. Check if there is any error in Discovery Startup.

#### Need additional checks?
Error Log in H2
```
2019-11-20 18:52:34.559 ERROR [-] [main] o.s.boot.SpringApplication               : Application startup failed
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'schedulingController': Unsatisfied dependency expressed through field 'scheduler'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'scheduler' defined in class path resource [app/metatron/discovery/config/SchedulingConfig.class]: Invocation of init method failed; nested exception is org.quartz.JobPersistenceException: Couldn't store job: Unique index or primary key violation: "QRTZ_TRIGGERS_IBFK_1_INDEX_B ON PUBLIC.QRTZ_JOB_DETAILS(SCHED_NAME) VALUES ('scheduler', 1)"; SQL statement:
INSERT INTO QRTZ_JOB_DETAILS (SCHED_NAME, JOB_NAME, JOB_GROUP, DESCRIPTION, JOB_CLASS_NAME, IS_DURABLE, IS_NONCONCURRENT, IS_UPDATE_DATA, REQUESTS_RECOVERY, JOB_DATA)  VALUES('scheduler', ?, ?, ?, ?, ?, ?, ?, ?, ?) [23505-196] [See nested exception: org.h2.jdbc.JdbcSQLException: Unique index or primary key violation: "QRTZ_TRIGGERS_IBFK_1_INDEX_B ON PUBLIC.QRTZ_JOB_DETAILS(SCHED_NAME) VALUES ('scheduler', 1)"; SQL statement:
INSERT INTO QRTZ_JOB_DETAILS (SCHED_NAME, JOB_NAME, JOB_GROUP, DESCRIPTION, JOB_CLASS_NAME, IS_DURABLE, IS_NONCONCURRENT, IS_UPDATE_DATA, REQUESTS_RECOVERY, JOB_DATA)  VALUES('scheduler', ?, ?, ?, ?, ?, ?, ?, ?, ?) [23505-196]]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:588)
        at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:88)
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:366)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1264)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:553)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:483)
        at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306)
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302)
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197)
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:761)
        at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:867)
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:543)
        at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:122)
```

### Types of changes



### Checklist:


### Additional Context
